gradle-plugin/src/main/java/com/contrastsecurity/gradle/plugin/ContrastGradlePlugin.java

```diff
@@ -11,11 +25,156 @@ public class ContrastGradlePlugin implements Plugin<Project> {
+    target.afterEvaluate(
+        project -> {
+          final ContrastSDK sdk = getSdkInstance(extension);
+
+          final TaskProvider<ResolveAgentTask> resolveAgentTask =
+              target.getTasks().register("resolveAgent", ResolveAgentTask.class, sdk);
+
+          resolveAgentTask.configure(
+              task -> {
+                task.getProvidedPath().set(extension.getJarPath());
+                task.getAgent().set(new File(agentPath));
+              });
+
+          // lifecycle task for grouping verification tasks
+          final TaskProvider<Task> contrastCheck = target.getTasks().register("contrastCheck");
+
+          final Collection<String> contrastArgs =
+              createContrastArgs(
+                  agentPath,
+                  extension.getAppName().get(),
+                  extension.getServerName().get(),
+                  extension.getAppVersion().get());
+
+          // for each test task...
+          target
+              .getTasks()
+              .withType(Test.class)
+              .forEach(
+                  testTask -> {
+                    testTask.dependsOn(resolveAgentTask);
+
+                    // attach agent args
+                    testTask.jvmArgs(contrastArgs);
+
+                    // generate ContrastVerifyTestTask for each test
+                    TaskProvider<ContrastVerifyTestTask> verifyTask =
+                        target
+                            .getTasks()
+                            .register(
+                                "contrastVerifyTest" + testTask.getName(),
+                                ContrastVerifyTestTask.class,
+                                v -> {
+                                  v.dependsOn(testTask);
+                                  // set file input for
+                                });
+
+                    // configure lifecycle task to depend on this test's corresponding verifyTask
+                    contrastCheck.configure(task -> task.dependsOn(verifyTask));
+                  });
+        });
+  }
+
+  /**
+   * Creates the jvmArgs for running the java agent against JavaExec tasks
+   *
+   * @param agentPath preconfigured path to an agent defined by the ContrastConfigurationExtension
+   * @param appName the name of the application on TS defined by the ContrastConfigurationExtension
+   * @param serverName the name of the server on TS defined by the ContrastConfigurationExtension
+   * @param appVersion the app version, if not supplied by the ContrastConfigurationExtension, then
+   *     it is generated in {@link ContrastGradlePlugin#computeAppVersion(String)}
+   * @return Set of arguments to be attached to specified tasks
+   */
+  @VisibleForTesting
+  public static Collection<String> createContrastArgs(
+      final String agentPath, final String appName, final String serverName, String appVersion) {
+
+    // List to preserve ordering of arguments
+    final Collection<String> args = new ArrayList<>();
```

Comment:
We can probably group these with `List.of` instead of adding one by one like this.